### PR TITLE
Fixes a few small bugs, general clean up to style

### PR DESCRIPTION
**Pull request by ZestyBread**
@Tratcher @BrennanConroy
Main things to look at:
VoidAction.cs (forgot to allow custom results)
UrlRewriteRule.cs
Culture invariance

### CODE REVIEW

File: src/Microsoft.AspNetCore.Rewrite/Internal/UrlRewrite/FileParser.cs [hunk 1]
```diff
@@ -168,10 +166,9 @@ private void ParseCondition(XElement condition, UrlRewriteRuleBuilder builder, P
 
             var parsedPatternString = condition.Attribute(RewriteTags.Pattern)?.Value;
 
-            Pattern input = null;
             try
             {
-                input = _inputParser.ParseInputString(parsedInputString);
+                var input = _inputParser.ParseInputString(parsedInputString);
                 builder.AddUrlCondition(input, parsedPatternString, patternSyntax, matchType, ignoreCase, negate);
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Rewrite/Internal/UrlRewrite/FileParser.cs]: nit: empty line

File: src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs [hunk 2]
```diff
@@ -69,6 +69,8 @@ public Task Invoke(HttpContext context)
                         return CompletedTask;
                     case RuleTerminiation.StopRules:
                         return _next(context);
+                    default:
+                        throw new ArgumentOutOfRangeException("Invalid rule termination.");
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs]: Include the value

**Comment by Tratcher** [comment]
:shipit:

File: src/Microsoft.AspNetCore.Rewrite/Internal/ModRewrite/Tokenizer.cs [hunk 3]
```diff
@@ -36,49 +37,50 @@ public List<string> Tokenize(string rule)
...
-                }
-                else if (context.Current == Space || context.Current == Tab)
-                {
-                    // time to capture!
-                    var token = context.Capture();
-                    if (!string.IsNullOrEmpty(token))
-                    {
-                        tokens.Add(token);
-                        while (context.Current == Space || context.Current == Tab)
+                        while (context.Current != Quote)
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Rewrite/Internal/ModRewrite/Tokenizer.cs]: nit: newline

File: src/Microsoft.AspNetCore.Rewrite/Internal/ModRewrite/Tokenizer.cs [hunk 4]
```diff
@@ -36,49 +37,50 @@ public List<string> Tokenize(string rule)
...
+                        }
+                        break;
+                    case Space:
+                    case Tab:
+                        // time to capture!
+                        var token = context.Capture();
+                        if (!string.IsNullOrEmpty(token))
+                        {
+                            tokens.Add(token);
+                            while (context.Current == Space || context.Current == Tab)
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Rewrite/Internal/ModRewrite/Tokenizer.cs]: do while :trollface:

File: src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/ConditionMatchSegment.cs [hunk 5]
```diff
@@ -5,16 +5,16 @@ namespace Microsoft.AspNetCore.Rewrite.Internal.PatternSegments
...
 
         public ConditionMatchSegment(int index)
         {
-            Index = index;
+            _index = index;
         }
 
         public override string Evaluate(RewriteContext context, MatchResults ruleMatch, MatchResults condMatch)
         {
-            return condMatch?.BackReference[Index]?.Value;
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/ConditionMatchSegment.cs]: Should the `?` be left?

**Pull request merged**